pennylane/grouping/pauli_group.py

```diff
@@ -0,0 +1,273 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Functions for generating and manipulating elements of the Pauli group.
+"""
+
+import numpy as np
+
+from pennylane import Identity
+from pennylane.grouping.utils import binary_to_pauli, pauli_to_binary, are_identical_pauli_words
+
+
+def pauli_group_generator(n_qubits, wire_map=None):
+    """Generator for iterating over the n-qubit Pauli group.
+
+    The :math:`n`-qubit Pauli group has size :math:`4^n`, thus it may not be desirable
+    to construct it in full and store. This function allows for iteration over elements
+    of the Pauli group with no storage involved.
+
+    The order of iteration is based on the binary symplectic representation of
+    the Pauli group as :math:`2n`-bit strings. Ordering is done by converting
+    the integers :math:`0` to :math:`2^{2n}` to binary strings, and converting those
+    strings to Pauli operators using the ``binary_to_pauli`` method.
+
+    Args:
+        n_qubits (int): The number of qubits for which to create the group.
+        wire_map (dict[Union[str, int], int]): dictionary containing all wire labels
+            used in the Pauli word as keys, and unique integer labels as their values.
+            If no wire map is provided, wires will be labeled by integers between 0 and ``n_qubits``.
+
+    Returns:
+        (qml.Operation): The next Pauli word in the group.
```

Comment:
Parenthesis aren't needed for return type, and the `qml` is just convention for PL - Sphinx won't be able to find it 🙂 
```suggestion
        .Operation: The next Pauli word in the group.
```